plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/utils/ReplicationControllerResourceProcessor.java

```diff
@@ -0,0 +1,102 @@
+
+    public ReplicationControllerResourceProcessor() {
+        super(true);
+    }
+    
+    @Override
+    public void handleDelete(ConnectionsRegistry registry, Connection connection, IResource resource, boolean willDeleteSubResources)
+            throws OpenShiftException {
+        IReplicationController rc = (IReplicationController)resource;
+
+        try {
+            if (willDeleteSubResources) {
+                List<IResource>[] overlappingRCs = findOverlappingReplicationControllers(connection, rc);
+                if (!overlappingRCs[1].isEmpty()) {
+                    String list = overlappingRCs[1].stream().map(res -> res.getName()).collect(Collectors.joining(","));
+                    throw new OpenShiftException(
+                            "Found overlapping replication controllers for replication controller %s: %s", rc.getName(),
+                            list);
+                } else if (overlappingRCs[0].size() == 1) {
+                    rc.setDesiredReplicaCount(0);
+                    rc = connection.updateResource(rc);
+                    while (rc.getCurrentReplicaCount() > 0) {
+                        Thread.sleep(1000);
+                        rc = connection.getResource(rc);
+                    }
+                } 
+            }
+            super.handleDelete(registry, connection, resource, willDeleteSubResources);
+        } catch (InterruptedException e) {
+            throw new OpenShiftException(e, e.getLocalizedMessage());
+        }
+    }
+
+    private List<IResource>[] findOverlappingReplicationControllers(Connection connection, IReplicationController replicationController) {
+        List[] overlappingRCS = new List[2];
+        overlappingRCS[0] = new ArrayList<>();
+        overlappingRCS[1] = new ArrayList<>();
+        List<IResource> rcs = connection.getResources(ResourceKind.REPLICATION_CONTROLLER, replicationController.getNamespace());
+        for(IResource rc : rcs) {
+            if (match((IReplicationController)rc, replicationController)) {
+                if (((IReplicationController)rc).getReplicaSelector().size() == replicationController.getReplicaSelector().size()) {
+                    overlappingRCS[0].add(rc);
+                } else {
+                    overlappingRCS[1].add(rc);
+                }
+            }
+        }
+        return overlappingRCS;
+    }
+    
+    private boolean match(IReplicationController rc, IReplicationController replicationController) {
+        return match(rc.getReplicaSelector(), replicationController.getReplicaSelector()) |
+               match(replicationController.getReplicaSelector(), rc.getReplicaSelector());
+    }
+
+    /**
+     * Find if maps matches. Maps matches if map1 is contained in map2
+     * @param map1 the first map
+     * @param map2 the second map
+     * @return the match state
+     */
+    private boolean match(Map<String, String> map1, Map<String, String> map2) {
```

Comment:
That is true, but it also should be called twice then. If match means that either of maps may contain the other, the new one-pass method may be added to ResourceUtils.